inc/ExampleApi/Queries/ExampleApiDataSource.php

```diff
@@ -2,13 +2,13 @@
 
 namespace RemoteDataBlocks\ExampleApi\Queries;
 
-use RemoteDataBlocks\Config\Datasource\HttpDatasource;
+use RemoteDataBlocks\Config\Datasource\DatasourceInterface;
 
 /**
  * This is a placeholder datasource used only to represent the data source in the
  * settings UI. The actual data loading is implemented by ExampleApiQueryRunner.
  */
-class ExampleApiDataSource extends HttpDatasource {
+class ExampleApiDataSource implements DatasourceInterface {
```

Comment:
👍 👍 

Yeah, that's the gist of it. Aligns the functionality in abstraction closer to true.